cctp_primitives/src/utils/serialization.rs

```diff
@@ -51,37 +90,23 @@ pub fn serialize_to_buffer<T: CanonicalSerialize>(
 
 pub const DEFAULT_BUF_SIZE: usize = 1 << 20;
 
-
 /// Deserialize from the file at `file_path` a compressed or uncompressed element,
 /// depending on the value of `compressed` flag, and perform checks on it, depending
 /// on the value of `semantic_checks` flag.
 /// `compressed` can be optional, due to some types being uncompressable;
 /// `semantic_checks` can be optional, due to some types having no checks to be performed,
 /// or trivial checks already performed a priori during serialization.
 pub fn read_from_file<T: CanonicalDeserialize + SemanticallyValid>(
```

Comment:
Can we get the problem similar to the one described in Issue 51 with this function? May be "read_from_file_strict" function should be created too?

Reply:
Sure, I was doing it but then we realized it is not needed anywhere and decided to postpone it

Reply:
Ok